wallet-new/integration/Functions.hs

```diff
@@ -436,27 +437,33 @@ runAction wc action = do
 
             -- Check the transaction.
             log $ "postTransaction: " <> show newPayment
-            result  <-  respToRes $ postTransaction wc newPayment
+            newTx  <-  respToRes $ postTransaction wc newPayment
+
+            let sumCoins f =
+                    sum . map (getCoin . unV1 . pdAmount) . toList $ f newTx
+                inputSum = sumCoins txInputs
+                outputSum = sumCoins txOutputs
 
-            let expectedAmount =
-                    V1 (mkCoin (getCoin moneyAmount - getCoin (unV1 (feeEstimatedAmount txFees))))
+            -- Transaction shouldn't produce any money
             checkInvariant
-                (txAmount result == expectedAmount)
-                (InvalidTransactionState result)
+                (inputSum >= outputSum)
```

Comment:
Isn't a stronger invariant `inputSum = outputSum + feeEstimatedAmount txFees`?

Reply:
Yes, second invariant is the same as the one you proposed. Additionaly your proposal doesn't use substraction so this check can be dropped then. Ty